org.opentosca.bus.management.api.osgievent/src/org/opentosca/bus/management/api/osgievent/route/Route.java

```diff
@@ -34,17 +35,17 @@ public void configure() throws Exception {
 
             exchange.getIn().setHeader(MBHeader.APIID_STRING.toString(), Activator.apiID);
 
-            final String messageID = exchange.getIn().getHeader("MessageID", String.class);
+            final String messageID =
+                exchange.getIn().getHeader(MBHeader.PLANCORRELATIONID_STRING.toString(), String.class);
             if (messageID != null) {
                 exchange.getIn().setMessageId(messageID);
-                exchange.getIn().removeHeader("MessageID");
-                exchange.getIn().setHeader(MBHeader.SYNCINVOCATION_BOOLEAN.toString(), "false");
+                exchange.getIn().setHeader(MBHeader.SYNCINVOCATION_BOOLEAN.toString(), false);
             } else {
-                exchange.getIn().setHeader(MBHeader.SYNCINVOCATION_BOOLEAN.toString(), "true");
+                exchange.getIn().setHeader(MBHeader.SYNCINVOCATION_BOOLEAN.toString(), true);
             }
 
-        }).to("stream:out").choice().when(this.header("OPERATION").isEqualTo("invokeIA")).to("direct:invokeIA")
-            .when(this.header("OPERATION").isEqualTo("invokePlan")).to("direct:invokePlan").end();
+        }).to("stream:out").choice().when(header("OPERATION").isEqualTo(Operations.invokeIA)).to("direct:invokeIA")
+            .when(header("OPERATION").isEqualTo(Operations.invokePlan)).to("direct:invokePlan").end();

```

Comment:
if this equality-check is the reason for the nonstandard naming of enum-members, consider exposing a member on the enums that specifically exposes the invoked name like the following:

```java
public enum Operations {
  INVOKE_PLAN("invokePlan"),
  INVOKE_IA("invokeIA");

  private final String headerValue;
  private Operations(String headerValue) {
    this.headerValue = headerValue;
  }
  public String getHeaderValue() {
    return headerValue;
  }
}

// use just like a normal member:
//[..]
.when(header("OPERATION").isEqualTo(Operations.INVOKE_PLAN.getHeaderValue()).to("direkt:invokePlan").end();
```